subsys/bluetooth/services/fast_pair/fp_battery.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/services/fast_pair.h>
+#include "fp_battery.h"
+
+static struct bt_fast_pair_battery_data fp_battery_data = {
+	.battery_comp_descs = {{ .charging = false,
+				 .level = BT_FAST_PAIR_BATTERY_COMP_DESC_LEVEL_UNKNOWN},
+			       { .charging = false,
+				 .level = BT_FAST_PAIR_BATTERY_COMP_DESC_LEVEL_UNKNOWN},
+			       { .charging = false,
+				 .level = BT_FAST_PAIR_BATTERY_COMP_DESC_LEVEL_UNKNOWN}}
+};
+
+static bool battery_comp_desc_valid(struct bt_fast_pair_battery_comp_desc battery_comp_desc)
+{
+	if ((battery_comp_desc.level > 100) &&
+	    (battery_comp_desc.level != BT_FAST_PAIR_BATTERY_COMP_DESC_LEVEL_UNKNOWN)) {
+		return false;
+	}
+
+	return true;
+}
+
+int bt_fast_pair_set_battery_comp_desc(enum bt_fast_pair_battery_comp battery_comp,
+				       struct bt_fast_pair_battery_comp_desc battery_comp_desc)
+{
+	if (!battery_comp_desc_valid(battery_comp_desc)) {
+		return -EINVAL;
+	}
```

Comment:
Maybe `is_battery_level_valid`? That's actually the only thing that we check here. We could pass only the value here too.